### PR TITLE
Python: Added metadata specifying connection stems from MSK code

### PR DESCRIPTION
**Pull request by Jibola**
### Description

* MongoDB drivers are used in various flavors and languages. Making sure we exercise our due diligence in identifying the "origin" of the library calls makes it best to understand how our Atlas servers get accessed. 
* Additionally, bumped the motor version to 3.3.2 to support the DriverInfo tracking behavior.

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

File: python/semantic_kernel/connectors/memory/mongodb_atlas/mongodb_atlas_memory_store.py [hunk 1]
```diff
@@ -44,6 +46,9 @@ def __init__(
         self._mongo_client = motor_asyncio.AsyncIOMotorClient(
             connection_string or mongodb_atlas_settings_from_dot_env(),
             read_preference=read_preference,
+            driver=DriverInfo(
+                "Microsoft Semantic Kernel", metadata.version("semantic-kernel")
```
Comment (nataschaZima) [inline python/semantic_kernel/connectors/memory/mongodb_atlas/mongodb_atlas_memory_store.py:50]: I need an example of a full driver_name string that results from this statement in order to be able to establish usage analytics when it goes live. Could anyone provide it?

Comment (Jibola) [reply python/semantic_kernel/connectors/memory/mongodb_atlas/mongodb_atlas_memory_store.py:50]: If I understand your request correctly: 
In this change, the driver name should come out something akin to: 
```
"driver_name": "PyMongo | Motor | Microsoft Semantic Kernel",
"version": "4.6 | 3.3.x | ...",
```

**Comment by moonbox3** [comment]
Hi @Jibola, is this work still needed? I am closing this PR as there has been no action since Dec 2023. If the work is still required, please re-open it with any required changes after pulling in main. Thank you.

File: python/semantic_kernel/connectors/memory/mongodb_atlas/mongodb_atlas_memory_store.py (hunk 1 above)
Comment (nataschaZima) [reply python/semantic_kernel/connectors/memory/mongodb_atlas/mongodb_atlas_memory_store.py:54]: ok this is sufficient, thank you

**Comment by Jibola** [comment]
> Hi @Jibola, is this work still needed? I am closing this PR as there has been no action since Dec 2023. If the work is still required, please re-open it with any required changes after pulling in main. Thank you.

Hi, it is ready. I've actually been waiting for review on the PR. I'll re-open.

**Pull request closed without merging**